src/sortStudents.ts

```diff
@@ -1,16 +1,62 @@
 
 export interface Student {
-  // describe Student interface
+  name: string,
+  surname: string,
+  age: number,
+  married: boolean,
+  grades: number[],
 }
 
 export enum SortType {
-  // describe SortType enum
+  Name = 'name',
+  Surname = 'surname',
+  Age = 'age',
+  Married = 'married',
+  AverageGrade = 'grades',
 }
 
-// create SortOrder type
-export type SortOrder;
+export type SortOrder = 'asc' | 'desc';
 
+export function getAverageGrades(arr: number[]): number {
+  const sum = arr.reduce((prev, curr) => prev + curr, 0);
 
-export function sortStudents(students, sortBy, order) {
-  // write your function
+  return sum / arr.length;
+}
+
+export function sortStudents(
+  students: Student[],
+  sortBy: SortType,
+  order: SortOrder,
+): Student[] {
+  const copyStudents = [...students];
+
+  switch (sortBy) {
+    case SortType.Name:
+    case SortType.Surname:
+      copyStudents.sort((a, b) => (order === 'asc'
+        ? a[sortBy].localeCompare(b[sortBy])
+        : b[sortBy].localeCompare(a[sortBy])
+      ));
+      break;
+
+    case SortType.Age:
+    case SortType.Married:
+      copyStudents.sort((a, b) => (order === 'asc'
+        ? +a[sortBy] - +b[sortBy]
```

Comment:
```suggestion
        ? Number(a[sortBy]) - Number(b[sortBy])
```
Number() is more obvious way